### ISSUE TITLE
Alignment incorrect for range declarations of unconstrained record elements spread over multiple lines

### ISSUE THREAD

**Issue by obruendl** [jeremiah-c-leary/vhdl-style-guide#1250]
**Environment**
Ubuntu 20.04

**Describe the bug**
Range declarations for unconstrained record elements are off by one column if spread over multiple lines (otherwise VSG does report errors). Can this be fixed? Example [here](https://github.com/open-logic/open-logic/blob/7b0ef64b3a840b31c8be15494833fc0903069acb/test/axi/olo_axi_master_full/olo_axi_master_full_tb.vhd#L66)

It's of course possible that this somehow has to do with my indent configuration. You can find my config file [here](https://github.com/open-logic/open-logic/blob/develop/lint/config/vsg_config.yml)

**To Reproduce**
1. Checkout Open-Logic (develop branch)
3. Run vsg on the file ("vsg -c ./lint/config/vsg_config.yml -f olo_axi_master_full_tb.vhd)

**Expected behavior**
I would expect all unconstrained record element range declarations to start on the same column but they are off by one. If I put them to the same column, this leads to linting errors.

**Screenshots**
![image](https://github.com/user-attachments/assets/a256ee6d-77b8-4a6c-b344-269774087217)

**Additional context**
None

**Comment by JHertz5** [comment]
Hi @obruendl. Thanks for raising this issue. I will try to take a look soon. It's a bit difficult to investigate this issue because the config file that you've referenced is > 4k lines long. If you would be able to provide a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example), i.e. a minimal config file that demonstrates your problem, it would help us to investigate this more quickly.

**Comment by obruendl** [comment]
See attached test case.

[test_case.zip](https://github.com/user-attachments/files/17463649/test_case.zip)

**Comment by JHertz5** [comment]
Hi @obruendl. Thanks very much for the MRE. I can reproduce your problem. In fact, I've discovered that it occurs with the default config, so it is not due to your config and there was no need to attach a config file after all. I'll investigate further when I get a chance and try to come up with a solution.

**Comment by JHertz5** [comment]
I've found the source of this problem. There is a variable called `bIgnoreStartParen`, which adds an extra space of indentation when enabled, so it appears that this was entirely deliberate. It is only enabled for the rules `signal_400` and `variable_400`. Here is an example of that code in action.

```python
                if bIgnoreStartParen:
                    dExpectedIndent[iLine + 1] = iColumn + 1
                else:
                    dExpectedIndent[iLine + 1] = iColumn
```

@jeremiah-c-leary There are no comments in the code that I can find to explain why this is being done. Do you have any recollection of why these rules have this extra indentation added? Would you be opposed to me removing the extra indentation to bring these rules in line with the rest of the rules?

**Comment by JHertz5** [comment]
After revisiting this, I can see that the formatting is broken in a different way if I just disable if statement and always use the second option. It will require further investigation.

**Comment by JHertz5** [comment]
I have raised a PR (#1345) to resolve this issue.
@obruendl Sorry for the delay! If you have a chance, please check that this branch works for you.

**Comment by jeremiah-c-leary** [comment]
Afternoon @JHertz5 ,

> @jeremiah-c-leary There are no comments in the code that I can find to explain why this is being done. Do you have any recollection of why these rules have this extra indentation added? Would you be opposed to me removing the extra indentation to bring these rules in line with the rest of the rules?

I used blame was tracked the addition to issue #539, but there was nothing specific about the `bIgnoreStartParen`.  It was added 3 years ago, so I do not remember exactly why it was added.  It could be to handle arrays, since the array would start with a parenthesis.

I will see if I can remember anything else.

--Jeremy

**Comment by JHertz5** [comment]
@obruendl pinging you again to check whether your happy with this fix?

**Comment by obruendl** [comment]
@JHertz5 

Sorry for the long delay, I was busy with some features and a conference.

Yes, the fix seems fine.

Could you ping me when it is released and let me know which  version it is? I will then upgrade to this version.

**Comment by JHertz5** [comment]
Hi @obruendl 

No problem, I have been very busy as well! Thanks for your feedback, I'll ask @jeremiah-c-leary to review and/or merge the branch.

> Could you ping me when it is released and let me know which version it is? I will then upgrade to this version.

I can't make any promises as I'm not keeping a close eye on the repo at the moment, but I'll try to remember.

Many thanks,
Jukka

**Comment by jeremiah-c-leary** [comment]
Morning @JHertz5 ,

Thanks for fixing this.

I will get this merged to master.

--Jeremy

**Comment by JHertz5** [comment]
Thanks very much @jeremiah-c-leary! @obruendl This has now been merged and will be in the next release.
